utils/fileutils.go

```diff
@@ -535,44 +522,41 @@ func Unmarshal(filePath string, loadTarget interface{}) (err error) {
 		return
 	}
 	defer func() {
-		closeErr := jsonFile.Close()
-		if err == nil {
-			err = closeErr
-		}
+		err = errors.Join(err, jsonFile.Close())
 	}()
 	var byteValue []byte
 	byteValue, err = io.ReadAll(jsonFile)
 	if err != nil {
 		return
 	}
-	return json.Unmarshal(byteValue, &loadTarget)
+	err = json.Unmarshal(byteValue, &loadTarget)
+	return
 }
 
 // strip '\n' or read until EOF, return error if read error
 // readNLines reads up to 'total' number of lines separated by \n.
 func ReadNLines(path string, total int) (lines []string, err error) {
 	reader, err := os.Open(path)
 	if err != nil {
-		return nil, err
+		return
 	}
 	defer func() {
-		deferErr := reader.Close()
-		if err == nil {
-			err = deferErr
-		}
+		err = errors.Join(err, reader.Close())
 	}()
 	r := bufio.NewReader(reader)
```

Comment:
"r" is a really bad name
```suggestion
	bufferedReader := bufio.NewReader(reader)
```